op-service/eth/blob.go

```diff
@@ -69,59 +71,198 @@ func VerifyBlobProof(blob *Blob, commitment kzg4844.Commitment, proof kzg4844.Pr
 }
 
 // FromData encodes the given input data into this blob. The encoding scheme is as follows:
-//
-// First, field elements are encoded as big-endian uint256 in BLS modulus range. To avoid modulus
-// overflow, we can't use the full 32 bytes, so we write data only to the topmost 31 bytes of each.
-// TODO: we can optimize this to get a bit more data from the blobs by using the top byte
-// partially.
-//
-// The first field element encodes the length of input data as a little endian uint32 in its
-// topmost 4 (out of 31) bytes, and the first 27 bytes of the input data in its remaining 27
-// bytes.
-//
-// The remaining field elements each encode 31 bytes of the remaining input data, up until the end
-// of the input.
-//
-// TODO: version the encoding format to allow for future encoding changes
+// It reads 31 bytes and then 1 byte for three times and then 31 bytes of data from the input.
+// For all the 31 bytes of data, they are encoded into [1:32] bytes of each field element.
+// For the extra 3 bytes read from the input, they are encoded into the 1 bytes of data on the top of each field element.
+// So for each field (4 field elements), 3 bytes are encoded into 4 bytes where the highest order bit is set to 0
```

Comment:
```suggestion
//
// In each round we perform 7 reads of input of lengths (31,1,31,1,31,1,31) bytes respectively for a total of 127
// bytes. This data is encoded into the next 4 field elements of the output by placing each of the 4x31
// byte chunks into bytes [1:32] of its respective field element. The three single byte chunks (24 bits) are
// split into 4x6-bit chunks, each of which is written into the top most byte of its respective field element,
// leaving the top 2 bits of each field element empty to avoid modulus overflow.
```